spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/EnvironmentEntryInitializingTreeMap.java

```diff
@@ -90,4 +90,8 @@ public T put(String key, T value) {
 		return delegate.entrySet();
 	}
 
+	@Override
+	public boolean containsKey(Object key) {
```

Comment:
Maybe this TreeMap will be used by another consumer in a near futur. In this case, the developer responsible for this new consumer will certainly not be aware of this treemap specific behaviour.

I think that this treemap should fully implement the map contract to avoid speculations on the consumers behaviours.